packages/site-admin/src/__core_unstable_copy__/packages/edit-site/src/components/sidebar-navigation-screen/index.tsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * External dependencies
+ */
+import {
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalHStack as HStack,
+
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalHeading as Heading,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { store as editSiteStore } from '@wordpress/edit-site';
+import { useContext } from '@wordpress/element';
+import { isRTL, __ } from '@wordpress/i18n';
+import { chevronRight, chevronLeft } from '@wordpress/icons';
+import clsx from 'clsx';
+/**
+ * Internal dependencies
+ */
+import { unstableResourceWarning } from '../../../../../debug';
+import { useHistory, useLocation } from '../../../../router/src';
+import { SidebarNavigationContext } from '../sidebar';
+import SidebarButton from '../sidebar-button';
+import './style.scss'; // @Todo: different from core: not imported in this way
+
+// import { privateApis as routerPrivateApis } from '@wordpress/router';
+// import { store as coreStore } from '@wordpress/core-data';
+// import {
+// 	isPreviewingTheme,
+// 	currentlyPreviewingTheme,
+// } from '../../utils/is-previewing-theme';
+// import { unlock } from '../../lock-unlock';
+
+// const { useHistory, useLocation } = unlock( routerPrivateApis );
+
+type SidebarNavigationScreenProps = {
+	isRoot: boolean;
+	title: string;
+	actions?: React.ReactNode;
+	meta?: React.ReactNode;
+	content: React.ReactNode;
+	footer?: React.ReactNode;
+	description?: string;
+	backPath?: string;
+};
+
+export default function SidebarNavigationScreen( {
+	isRoot,
+	title,
+	actions,
+	meta,
+	content,
+	footer,
+	description,
+	backPath: backPathProp,
+}: SidebarNavigationScreenProps ) {
+	unstableResourceWarning(
+		'<SidebarNavigationScreen />',
+		'https://github.com/WordPress/gutenberg/blob/56883338749aa23acc75481c3bbc605bf1cb5a81/packages/edit-site/src/components/sidebar-navigation-screen/index.js#L35'
+	);
+
+	// previewingThemeName
+	const { dashboardLink, dashboardLinkText } = useSelect( ( select ) => {
+		const { getSettings } = select( editSiteStore );
+		// const currentlyPreviewingThemeId = currentlyPreviewingTheme();
+		return {
+			dashboardLink: getSettings().__experimentalDashboardLink,
+			dashboardLinkText: getSettings().__experimentalDashboardLinkText,
```

Comment:
It seems like the only reason for the editSite dependency is the "dashboard link and text". I have two thoughts:

 1- Ultimately, these can be potentially moved to the `/`  (root) REST API endpoint in Core instead of being just edit-site settings.
 2- Potentially for the sake of this package, we can just have a "onNavigateBack" prop or something like that.

Reply:
Yes, something similar I did for the SiteHubMobile component [here](https://github.com/Automattic/wp-calypso/blob/dd56a7003ba5b6b59f534467d2b36d43a9d01a22/packages/site-admin/src/__core_unstable_copy__/packages/edit-site/src/components/site-hub/index.tsx#L127)